### ISSUE TITLE
Optimizer wrapper performance is dependent on tensorflow version

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#96]
**Describe the bug**
It has been observed that for `tf > 2.10`, the approximation performance is poorer than for older versions.
This was mentioned [here](https://github.com/andreped/GradientAccumulator/blob/tf-fix/tests/test_expected_result.py#L106).

**To Reproduce**
Modify the test script below [here](https://github.com/andreped/GradientAccumulator/blob/main/tests/test_expected_result.py) to enforce checks with `result1 == result2` instead of the numpy almost equal checks.
```
python3 -m pytest -v tests/test_expected_result.py
```

**Expected behavior**
The optimizer wrapper should have identical performance for all supported tf versions.

**Comment by andreped** [comment]
As `enable_op_determinism` was introduced in `tf==2.7`, tests on older tf versions can be unstable, as different results can be produced for the same setup, as we are unable to properly reset and set seeds between runs.
```
module 'tensorflow._api.v2.config.experimental' has no attribute 'enable_op_determinism'
```

An idea is thus to allow poorer approximation between experiments for older tf versions.

**Comment by andreped** [comment]
> An idea is thus to allow poorer approximation between experiments for older tf versions.

Added fix for this issue in c314adaf15e0c014d78780ef698d29830fe5b1d4.

**Comment by andreped** [comment]
> > An idea is thus to allow poorer approximation between experiments for older tf versions.
> 
> Added fix for this issue in [c314ada](https://github.com/andreped/GradientAccumulator/commit/c314adaf15e0c014d78780ef698d29830fe5b1d4).

This had already been fixed above, but I had forgotten to close the issue. I am therefore closing it now.
